torch/lib/c10d/ProcessGroupNCCL.cpp

```diff
@@ -1159,12 +1160,20 @@ c10::intrusive_ptr<ProcessGroup::Work> ProcessGroupNCCL::pointToPoint(
   syncStreams(devices, ncclEvents_[key], ncclStreams_[key]);
 
   // Work itself will create the CUDA events on all GPUs of tensors
-  auto work = initWork(devices, rank_, opType);
+  bool can_profile = tensors.size() == 1;
```

Comment:
shall we add some comments to explain why we have this restriction?

Reply:
This is because when https://github.com/pytorch/pytorch/pull/46471/files that originally enabled the profiling was added, we used NCCLFuture instead of CUDAFuture which only supported single device per process. 

Now that we use CUDAFuture it might be possible this restriction can be removed. Will add comment and file issue for that.

Reply:
https://github.com/pytorch/pytorch/issues/56836